### PR TITLE
Fix --no-default-features

### PR DESCRIPTION
**Pull request by c410-f3r**
`cargo build/check/test --no-default-features` is broken and a CI step was included to avoid the same problem in the future

### CODE REVIEW

File: src/listener/tcp_listener.rs [hunk 1]
```diff
@@ -87,8 +91,9 @@ impl<State: Clone + Send + Sync + 'static> Listener<State> for TcpListener {
                     continue;
                 }
 
-                Ok(stream) => {
-                    handle_tcp(app.clone(), stream);
+                Ok(_stream) => {
+                    #[cfg(feature = "h1-server")]
+                    handle_tcp(_app.clone(), _stream);
```
Comment (c410-f3r) [inline src/listener/tcp_listener.rs:96]: This is probably not the right solution. Any better ideas?

Comment (jbr) [reply src/listener/tcp_listener.rs:96]: I think there's no reason to include any of the Listener implementations unless h1-server is provided

Comment (jbr) [reply src/listener/tcp_listener.rs:96]: Addressed this in #678

**Comment by jbr** [comment]
Closing in preference to #678, which makes more drastic removals for --no-default-features.  Thanks for raising this issue!

**Comment by c410-f3r** [comment]
This closure was unexpected. The work was indented to fix the problem and if the the current approach isn't desired, then you guys could at least hint the appropriated direction.

**Comment by jbr** [comment]
I'm sorry! I didn't want to make more work for you by saying "please re-do this pr." Fixing --no-default-features required making a lot of listener code conditional (everything except the traits)

**Comment by c410-f3r** [comment]
@jbr Dont worry. Thanks for the fix 👍

**Pull request closed without merging**